<?xml version="1.0"?>
<launch>
    <arg name="gui" default="false" />
    <arg name="rviz" default="true" />

    <param name="ROBOT_NAME" value="$(env ROBOT_NAME)"/>
    <param name="ROBOT_SDF_FILE" value="$(find robot_fleet)/static_models_w_plugin/husky/model.sdf"/>
    
    <arg name="ROSBRIDGE_STATE" value="$(env ROSBRIDGE_STATE)" />
    <param name="ROSBRIDGE_STATE" value="$(env ROSBRIDGE_STATE)"/>
    <param name="ROSBRIDGE_IP" value="$(env ROSBRIDGE_IP)" />

    <param name="use_custom_move_object_gazebo_plugin" value="$(optenv USE_CUSTOM_MOVE_OBJECT_GAZEBO_PLUGIN false)" />

    <group if="$(eval ROSBRIDGE_STATE=='SERVER')" >
        <param name="ROSBRIDGE_IP" value="localhost"/>
        <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />
    </group>
    
    <!-- Launch robot_fleet specific nodes -->
    <node name="gazebo_model_mover" pkg="robot_fleet" type="gazebo_model_mover.py" />
    <node name="client_rosbridge" pkg="robot_fleet" type="client_rosbridge.py" />

    <!--- Run husky app -->
    <include file="$(find robot_fleet)/launch/husky_app_only.launch" />


</launch>
